CMakeLists.txt

```diff
@@ -88,6 +92,39 @@ endif ()
 
 if (USE_OPENMP)
   find_package(OpenMP REQUIRED)
+
+  # necessarily a starting point; ultimately the OpenMP module/package must to be extended
+  if (USE_OFFLOAD AND WITH_OPENMP_FLAGS)
```

Comment:
If I get the logic correctly, you're bypassing CMake's checks of the given flags here: The user can get a "OpenMP found - everything's fine" from CMake, but if he passes invalid flags the compilation fails later on.

At least for the `WITH_OPENMP_FLAGS` it should be possible to set `OpenMP_${LANG}_FLAG` prior to the `find_package`, if we can't tell the user to directly set `OpenMP_${LANG}_FLAG` (which I'd prefer rather than introducing another custom flag).

Reply:
Both flags serve a temporary purpose:

* WITH_OFFLOAD_FLAGS: is meant to add an additional flag to the list of OpenMP flags, e.g., `-qoffload`. This is not supported yet by the CMake OpenMP module.
* WITH_OPENMP_FLAGS: is meant to replace OpenMP flags regardless of what was figured out by the OpenMP CMake-module. This serves for instance Intel's OneAPI compiler for the time being. At the moment there special flags needed to trigger the OpenMP based GPU code. 

Anyhow, both of these helper options are nothing to worry about. This can be easily cleaned up, and should not interfere.